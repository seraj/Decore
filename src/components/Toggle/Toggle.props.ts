import CheckboxProps from "../Checkbox/Checkbox.props";

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export default interface ToggleProps extends CheckboxProps {}
