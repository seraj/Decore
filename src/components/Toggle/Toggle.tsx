import * as React from "react";
import StyledToggle from "./styled/StyledToggle";
import ToggleProps from "./Toggle.props";
import {
  HiddenCheckbox,
  CheckboxContainer
} from "../Checkbox/styled/StyledCheckbox";

const Toggle: React.FC<ToggleProps> = ({
  checked,
  disabled,
  className,
  children,
  ...props
}) => (
  <CheckboxContainer className={className}>
    <HiddenCheckbox
      {...props}
      type="checkbox"
      checked={checked}
      disabled={disabled}
    />
    <StyledToggle {...props} checked={checked} disabled={disabled} />
  </CheckboxContainer>
);

export default Toggle;
