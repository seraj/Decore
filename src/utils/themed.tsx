import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
