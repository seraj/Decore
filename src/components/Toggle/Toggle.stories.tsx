import { boolean, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Toggle from "./Toggle";

const stories = storiesOf("Components/Toggle", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => (
    <Toggle
      checked={boolean("checked", false)}
      disabled={boolean("disabled", false)}
    />
  ))
);
