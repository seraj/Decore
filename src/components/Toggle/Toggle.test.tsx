import * as React from "react";

import themed from "../../utils/themed";
import Toggle from "./Toggle";

describe("Toggle", () => {
  it("should render correctly", () => {
    const component = themed(<Toggle>Test</Toggle>).toJSON();

    expect(component).toMatchSnapshot();
  });
});
