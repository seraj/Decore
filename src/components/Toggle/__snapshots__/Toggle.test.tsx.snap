// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should render correctly 1`] = `
.c3 {
  display: inline-block;
  width: 58px;
  height: 29px;
  background: #ffffff;
  border-color: #999999;
  border-width: 1.5px;
  border-style: solid;
  position: relative;
}

.c3:hover {
  border-color: #666666;
}

.c3::after {
  content: 'Off';
  font-size: 0.6em;
  font-weight: bold;
  line-height: 22px;
  color: #ffffff;
  text-align: center;
  text-transform: uppercase;
  background: #999999;
  width: 25px;
  height: 22px;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  position: absolute;
  left: 2px;
  top: 2px;
  display: inline-block;
  -webkit-transition: 250ms;
  transition: 250ms;
}

.c2 {
  border: 1.5px solid #999999;
  background-color: undefined;
  padding: 1rem 0.8rem;
  height: 45px;
  border-radius: 0px;
  color: #333333;
}

.c2:hover {
  border-color: #333333;
}

.c2:disabled {
  background: #E5E5E5;
  color: #999999;
  cursor: not-allowed !important;
  pointer-events: none;
}

.c0 {
  display: inline-block;
  vertical-align: middle;
}

.c1 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clippath: inset(50%);
  clippath: inset(50%);
  height: 1px !important;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px !important;
}

<div
  className="c0"
>
  <input
    className="c1 c2"
    type="checkbox"
  />
  <div
    className="c3"
  />
</div>
`;
